gdal/frmts/netcdf/netcdfdataset.cpp

```diff
@@ -4849,48 +4868,50 @@ double netCDFDataset::rint( double dfX )
 CPLErr netCDFDataset::ReadAttributes( int cdfidIn, int var)
 
 {
-    char szVarName[NC_MAX_NAME + 1];
-    int nbAttr = 0;
+    char *pszVarFullName = nullptr;
+    ERR_RET(NCDFGetVarFullName(cdfidIn, var, &pszVarFullName));
+    size_t nMetaNameSize = sizeof(char) * (strlen(pszVarFullName) + 1
```

Comment:
check pszVarFullName != nullptr ?

Reply:
ERR_RET() returns if NCDFGetVarFullName() fails so pszVarFullName = nullptr should never be used if I suppose that NCDFGetVarFullName() is well implemented (it should never returns CE_None if pszVarFullName=nullptr) but if you prefer I can handle this case. I think I have other cases like this, for example in NCDFGetVarFullName().

Reply:
Fine if ERR_RET() and NCDFGetVarFullName() provide this guarantee.